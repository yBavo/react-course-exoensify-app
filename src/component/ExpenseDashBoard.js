import React from 'react';
import ExpenseListFilter from './ExpenseListFilters'
import ExpenseList from './ExpenseList'


const ExpenseDashBoardPage = () => (
	<div>
		<h1>Expense DashBoard</h1>
		<div>
		<ExpenseListFilter />
		<ExpenseList />
		</div>
	</div>
);

export default ExpenseDashBoardPage;
