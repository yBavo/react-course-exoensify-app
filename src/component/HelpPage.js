import React from 'react';

const HelpPage = () => (
	<div>
		<h1>Help Page</h1>
	</div>
);

export default HelpPage;
