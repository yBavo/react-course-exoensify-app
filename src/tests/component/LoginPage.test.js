import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { LoginPage } from '../../component/LoginPage';
//import { startLogin } from '../../actions/auth';

configure({ adapter: new Adapter() });

let startLogin, wrapper;
beforeEach(() => {
  startLogin = jest.fn();
  wrapper = shallow(
    <LoginPage startLogin={startLogin} />
  );
})

test('Should render Login page correctly', () => {
  expect(wrapper).toMatchSnapshot();
})

test('Should call startLogin on button click', () => {
  wrapper.find('button').simulate('click');
  expect(startLogin).toHaveBeenCalled()
})