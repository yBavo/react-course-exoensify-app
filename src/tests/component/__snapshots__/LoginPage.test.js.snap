// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Login page correctly 1`] = `
<div>
  <button
    onClick={[MockFunction]}
  >
    Login
  </button>
</div>
`;
